crates/nu-std/lib/log.nu

```diff
@@ -56,52 +145,129 @@ def now [] {
+export def "log warning" [
+    message: string, # A message
+    --short (-s) # Whether to use a short prefix
+] {
+    if (current-log-level) > (log WARNING_LEVEL) {
+        return
+    }
+
+    let prefix = (if $short {
+	    log WARNING_LEVEL_PREFIX --short
+    } else {
+        log WARNING_LEVEL_PREFIX
+    })
+    log-formatted (ansi yellow) $prefix $message
 }
-# Log info message
-export def "log info" [message: string] {
-    if (current-log-level) > (log INFO_LEVEL) { return }
 
-    print --stderr $"(ansi default)(log INFO_LEVEL_PREFIX)|(now)|($message)(ansi reset)"
+# Log an info message
+export def "log info" [
+    message: string, # A message
+    --short (-s) # Whether to use a short prefix
+] {
+    if (current-log-level) > (log INFO_LEVEL) {
+        return
+    }
+
+    let prefix = (if $short {
+	    log INFO_LEVEL_PREFIX --short
+    } else {
+        log INFO_LEVEL_PREFIX
+    })
+    log-formatted (ansi default) $prefix $message
 }
-# Log debug message
-export def "log debug" [message: string] {
-    if (current-log-level) > (log DEBUG_LEVEL) { return }
 
-    print --stderr $"(ansi default_dimmed)(log DEBUG_LEVEL_PREFIX)|(now)|($message)(ansi reset)"
+# Log a debug message
+export def "log debug" [
+    message: string, # A message
+    --short (-s) # Whether to use a short prefix
+] {
+    if (current-log-level) > (log DEBUG_LEVEL) {
+        return
+    }
+
+    let prefix = (if $short {
+	    log DEBUG_LEVEL_PREFIX --short
+    } else {
+        log DEBUG_LEVEL_PREFIX
+    })
+    log-formatted (ansi default_dimmed) $prefix $message
 }
 
-# Log with custom message format and verbosity level
+# Log a message with a specific format and verbosity level
 # 
 # Format reference:
-#   > %MSG% will be replaced by $message
-#   > %DATE% will be replaced by the timestamp of log in standard Nushell's log format: "%Y-%m-%dT%H:%M:%S%.3f"
-#   > %LEVEL% will be replaced by the standard Nushell's log verbosity prefixes, e.g. "CRT"
+# - %MSG% will be replaced by $message
+# - %DATE% will be replaced by the timestamp of log in standard Nushell's log format: "%Y-%m-%dT%H:%M:%S%.3f"
+# - %LEVEL% will be replaced by the standard Nushell's log verbosity prefixes, e.g. "CRT"
 #
 # Examples:
-#   > std log custom "my message" $"(ansi yellow)[%LEVEL%]MY MESSAGE: %MSG% [%DATE%](ansi reset)" (std log WARNING_LEVEL) 
+# - std log custom "my message" $"(ansi yellow)[%LEVEL%]MY MESSAGE: %MSG% [%DATE%](ansi reset)" (std log WARNING_LEVEL)
 export def "log custom" [
-    message: string, # Message inserted into the log template
-    format: string, # A string to be printed into stderr. 
-    log_level: int # Log's verbosity level
-    ] {
-    if (current-log-level) > ($log_level) { return }
+    message: string, # A message
+    format: string, # A format
+    log_level: int # A log level
+    --short (-s) # Whether to use a short prefix
+] {
+    if (current-log-level) > ($log_level) {
+        return
+    }
+
+    mut level = ((if $short {
```

Comment:
Why `mut`? At first glance it seems like we're not modifying it anywhere further

Reply:
Fixed. ;)